hazelcast-ra/hazelcast-jca/src/test/java/com/hazelcast/jca/XATestWithJCA.java

```diff
@@ -79,9 +82,76 @@ public void testPut() throws ResourceException {
         assertEquals(val, m.get(key));
     }
 
+    @Test
+    public void testTransactionalQueueShouldPollWhatWasOffered() throws ResourceException {
+        String name = randomString();
+        String item = randomString();
+
+        TransactionalQueue<String> queue = connection.getTransactionalQueue(name);
+        queue.offer(item);
+
+        connection.close();
+
+        IQueue<String> q = instance.getQueue(name);
+        assertEquals(item,q.poll());
+    }
+
+    @Test
+    public void testTransactionalMultiMapShouldHaveWhatIsInserted() throws ResourceException {
+        String name = randomString();
+        String key = randomString();
+        String val1 = randomString();
+        String val2 = randomString();
+
+        TransactionalMultiMap<String, String> multiMap = connection.getTransactionalMultiMap(name);
+        multiMap.put(key,val1);
+        multiMap.put(key,val2);
+
+        connection.close();
+
+        MultiMap<String, String> m = instance.getMultiMap(name);
+        Collection<String> vals = m.get(key);
+        assertTrue(vals.contains(val1));
+        assertTrue(vals.contains(val2));
+        assertTrue(vals.size() == 2);
```

Comment:
+1. The advantage is that in case of failure, you know exactly what the values were. With assertTrue you loose that information unless you craft a message.